internal/platform/platform.go

```diff
@@ -0,0 +1,387 @@
+	stateEcefVZ := getDatasetFloat32(filename, group, "acoOnGnssStateEcef_vz")
+	stateEcefX := getDatasetFloat32(filename, group, "acoOnGnssStateEcef_x")
+	stateEcefY := getDatasetFloat32(filename, group, "acoOnGnssStateEcef_y")
+	stateEcefZ := getDatasetFloat32(filename, group, "acoOnGnssStateEcef_z")
+	stateTime := getDatasetFloat64(filename, group, "acoOnGnssStateTime")
+
+	var records []GnssRecord
+	for i := range time {
+		record := GnssRecord{
+			Time:            toDateTime(time[i]),
+			PropagationTime: propagationTime[i],
+			StateEcefVX:     stateEcefVX[i],
+			StateEcefVY:     stateEcefVY[i],
+			StateEcefVZ:     stateEcefVZ[i],
+			StateEcefX:      stateEcefX[i],
+			StateEcefY:      stateEcefY[i],
+			StateEcefZ:      stateEcefZ[i],
+			StateTime:       stateTime[i],
+		}
+		records = append(records, record)
+	}
+	return records
+
+}
+
+func getOrbitRecords(filename string) []OrbitRecord {
+
+	var group string = "root/Level1A/ReconstructedData/PreciseOrbitEstimation"
+
+	time := getDatasetFloat64(filename, group, "Time")
+	gnssStateJ2000 := getDatasetFloat64(filename, group, "acsGnssStateJ2000")
+	uncertainty := getDatasetFloat64(filename, group, "acsNavigationUncertainty")
+
+	//uncertainty is common to all records
+	uncertainty2d := to6by6arr(uncertainty)
+
+	var records []OrbitRecord
+	n := len(time)
+	for i := range time {
+		record := OrbitRecord{
+			Time:        toDateTime(time[i]),
+			Uncertainty: uncertainty2d,
+		}
+		copy(record.GnssStateJ2000[:], to1DSlice(gnssStateJ2000, 6, n, i))
+		records = append(records, record)
+	}
+	return records
+}
+
+func getAttitudeRecords(filename string) []AttitudeRecord {
+
+	var group string = "root/Level1A/ReconstructedData/PreciseAttitudeEstimation"
+
+	time := getDatasetFloat64(filename, group, "Time")
+	attitudeState := getDatasetFloat64(filename, group, "afsAttitudeState")
+	attitudeUncertainty := getDatasetFloat64(filename, group, "afsAttitudeUncertainty")
+	rateUncertainty := getDatasetFloat64(filename, group, "afsRateUncertainty")
+	spacecraftRate := getDatasetFloat64(filename, group, "afsSpacecraftRate")
+	tangentPoint := getDatasetFloat64(filename, group, "afsTangentPoint")
+
+	var records []AttitudeRecord
+	n := len(time)
+	for i := range time {
+		record := AttitudeRecord{Time: toDateTime(time[i])}
+		copy(record.SpacecraftRate[:], to1DSlice(spacecraftRate, 3, n, i))
+		copy(record.TangentPoint[:], to1DSlice(tangentPoint, 3, n, i))
+		copy(record.AttitudeState[:], to1DSlice(attitudeState, 4, n, i))
+		record.AttitudeUncertainty = to3by3arr(attitudeUncertainty, i)
+		record.RateUncertainty = to3by3arr(rateUncertainty, i)
+		records = append(records, record)
+	}
+	return records
+}
+
+func to1DSlice(inslice []float64, sizeDim1 int, sizeDim2, offset int) []float64 {
+	//returns a 1-d slice extracted from a 1-d slice that is a representation
+	//of a 2-Darray with shape (sizeDim1, sizeDim2)
+	outslice := make([]float64, sizeDim1)
+	for i := 0; i < sizeDim1; i++ {
```

Comment:
I suppose you could use `range` here if you like.

Reply:
range works on slices not on int

Reply:
I meant `for i := range outslice` but maybe that doesn't work?

Reply:
```suggestion
	for i := range outslice {
```